### PR TITLE
NetCore: Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by elit0451**
Creating a YAML pipeline for Azure DevOps for .NET Core project.

It includes:
- executing Unit Tests on Linux image;
- executing Unit Tests on Windows image + building the NuGet packages;
- executing Integration Tests on Windows (for now all tests are running on 1 server).

### CODE REVIEW

File: build/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  - job: Unit_Tests
+    displayName: 'Unit Tests'
+    pool:
+      vmImage: 'ubuntu-latest'
+    steps:
+  
+    - task: UseDotNet@2
+      displayName: 'Use .Net Core sdk 3.1.100'
+      inputs:
+        version: 3.1.100
```
Comment (JimBobSquarePants) [inline build/azure-pipelines.yml:25]: That SDK version is out of date. Use 3.1.x to automatically get the latest.

Comment (elit0451) [reply build/azure-pipelines.yml:25]: I'll update, thanks

**Comment by JimBobSquarePants** [comment]
That test error seems to indicate that the path the TypeLoader is attempting to access is not fully qualified. 

```
X Can_Build_ContentTypeAlias_Query_For_IContentType [< 1ms]
  Error Message:
   System.UnauthorizedAccessException : Access to the path '/App_Data/TEMP/TypesCache' is denied.
  ----> System.IO.IOException : Permission denied
  Stack Trace:
     at System.IO.FileSystem.CreateDirectory(String fullPath)
   at System.IO.Directory.CreateDirectory(String path)
   at Umbraco.Core.Composing.TypeLoader.GetFileBasePath() in /home/vsts/work/1/s/src/Umbraco.Core/Composing/TypeLoader.cs:line 306
   at Umbraco.Core.Composing.TypeLoader.GetTypesHashFilePath() in /home/vsts/work/1/s/src/Umbraco.Core/Composing/TypeLoader.cs:line 290
   at Umbraco.Core.Composing.TypeLoader.get_CachedAssembliesHash() in /home/vsts/work/1/s/src/Umbraco.Core/Composing/TypeLoader.cs:line 162
   at Umbraco.Core.Composing.TypeLoader..ctor(ITypeFinder typeFinder, IAppPolicyCache runtimeCache, DirectoryInfo localTempPath, ILogger`1 logger, IProfilingLogger profilingLogger, Boolean detectChanges, IEnumerable`1 assembliesToScan) in /home/vsts/work/1/s/src/Umbraco.Core/Composing/TypeLoader.cs:line 82
   at Umbraco.Core.Composing.TypeLoader..ctor(ITypeFinder typeFinder, IAppPolicyCache runtimeCache, DirectoryInfo localTempPath, ILogger`1 logger, IProfilingLogger profilingLogger, IEnumerable`1 assembliesToScan) in /home/vsts/work/1/s/src/Umbraco.Core/Composing/TypeLoader.cs:line 57
   at Umbraco.Tests.Common.TestHelperBase.GetMockedTypeLoader() in /home/vsts/work/1/s/src/Umbraco.Tests.Common/TestHelperBase.cs:line 51
   at Umbraco.Tests.TestHelpers.TestHelper.GetMockedTypeLoader() in /home/vsts/work/1/s/src/Umbraco.Tests.UnitTests/TestHelpers/TestHelper.cs:line 87
   at Umbraco.Tests.TestHelpers.BaseUsingSqlSyntax.Setup() in /home/vsts/work/1/s/src/Umbraco.Tests.UnitTests/TestHelpers/BaseUsingSqlSyntax.cs:line 35
--IOException

```

https://dev.azure.com/umbraco/Umbraco%20Cms/_build/results?buildId=52724&view=logs&j=df2de58d-0893-5ed6-6d94-5b99c093de3e&t=e5fe5dda-854d-5fa1-d6dd-0aa45c163423&l=1863

**Comment by JimBobSquarePants** [comment]
Hurrah!

**Pull request merged**